### PR TITLE
Use `pytest`

### LINKED ISSUE #47: Get testing with pytest
**Is your feature request related to a problem? Please describe.**
No.

**Describe the solution you'd like**
Testing with pytest makes everything quicker, more efficient, and faster.

**Describe alternatives you've considered**
Running each command manually, which is less efficient, and reports aren't given, unlike pytest, which files error and warning reports in the summary.

### PR DESCRIPTION
**Pull request by quantum9Innovation**
# Set up `pytest` as main testing framework
After PR #46, solves issue #47 
___

## New Feature
Transforms old tests to use `pytest` as the main testing framework.
`pytest` allows for
- more testing accuracy
- easily accessible testing suite
- fast testing due to local cache
- `assert`-based testing and bounds
It also adds GitHub CODEOWNERS
## Known Issues
Fixes #47 "Getting started with pytest"
## Code Breakdown
 - .github/CODEOWNERS
     - Adds CODEOWNERS
 - .github/workflows/
     - build.yml, coverage.yml
         - Changed testing to use `$ python -m pytest` command
     - Optimizes workflows (@Quantalabs check optimizations)
 - bin/epispot
     - Replace broad `except` with `except ImportError`
 - bin/requirements.txt
     - Removes `epispot` due to `pip install -r bin/requirements.txt` command in workflows which installs epispot from PyPI instead of from the source
 - epispot/__init__.py
     - Add shortened README and guide to top of file (for `help()` commands)
     - Add dependency and sanity checks (@Quantalabs please see)
     - Remove `__` from metadata variable names to make them globally accessible
     - Add documentation on metadata
 - epispot/
     - Add `# pragma: no cover` on soon-to-be deprecated functions
- explorables/
    - Remove all tests, former testing suites, examples, and tutorials and move them to this directory to keep `tests/` directory clean
- tests/
    - Replace testing suite with new one that uses `pytest`
- release.py
    - Update to be compatible with new `__init__.py`
- setup.py, setup-nightly.py
    - Add more metadata and classifier information
## Additional Notes
There's a lot going on in this PR and it may need to be broken up into multiple PRs.
For now, it will be created as a draft PR just because of the sheer scale of the PR.
Additionally, before moving into merge stage, all merge conflicts should be resolved and all basic checks (build and code coverage) should pass. After we enter the merge stage, reviewers should analyze code for compatibility issues and other code quality-related alerts will be dealt with if necessary.

---
Closes #47

### CODE REVIEW

**Comment by quantum9Innovation** [comment]
Code coverage failed due to not testing the CLI, which will be fixed in a later PR. For now, code coverage is essentially *passing*.
Opening for review after:
- LGTM Analysis (exit code irrelevant :white_check_mark: )
- CodeQL Analysis (must pass :heavy_check_mark: )

**Comment by quantum9Innovation** [comment]
We are now ready for review, pinging @Quantalabs

File: .github/workflows/vulnerability.yml [hunk 1]
```diff
@@ -8,15 +8,6 @@ name: Vulnerability Scope Analyzer
 # or API.
```
Comment (Quantalabs) [inline .github/workflows/vulnerability.yml:8]: Workflow builds are public, we shouldn't have this file at all, as it displays all the environment variables, posing a security vulnerability already.

File: tests/test_advanced.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    SIRS_Model.add_layer(Susceptible, 'Susceptible', [Infected])
+    SIRS_Model.add_layer(Infected, 'Infected', [Removed])
+    SIRS_Model.add_layer(Removed, 'Removed', [Susceptible])
+
+    # solve
+    Solution = SIRS_Model.integrate(range(100), starting_state=[0.9 * N(0), 0.1 * N(0), 0])
+    return Solution
+
+
+def test_SIHCR():
```
Comment (Quantalabs) [inline tests/test_advanced.py:86]: You should put `test_SIHCR` and `test_SIRS` in a test class.

**Review by Quantalabs — CHANGES_REQUESTED** [review]

File: tests/test_advanced.py (hunk 2 above)
Comment (quantum9Innovation) [reply tests/test_advanced.py:86]: What does a test class do and what benefits does it offer?

**Comment by quantum9Innovation** [comment]
After review, we're splitting up the PR into multiple PRs that will be merged in a certain order specified on this PR and the Management project

File: .github/workflows/vulnerability.yml (hunk 1 above)
Comment (quantum9Innovation) [reply .github/workflows/vulnerability.yml:8]: Done in `testing-workflows`

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -21,10 +21,14 @@ jobs:
...
+        pip install -U pytest
+        pip install -U coverage
+        pip install -U codecov
         pip install -r requirements.txt
+        pip install -r bin/requirements.txt
+    - name: Build from source
+      run: |
+        python setup-nightly.py install
     - name: Prepares Code Coverage Report 
       continue-on-error: true
```
[the remark this answers is not in the document]
Comment (quantum9Innovation) [reply .github/workflows/coverage.yml:33]: Done in `testing-workflows`

**Pull request closed without merging**